src/com/moclojer/rq/queue.clj

```diff
@@ -1,73 +1,16 @@
 (ns com.moclojer.rq.queue
   (:refer-clojure :exclude [pop!])
   (:require
-   [clojure.edn :as edn]
-   [clojure.tools.logging :as log]
-   [com.moclojer.rq.utils :as utils]))
-
-(defn push!
```

Comment:
@J0sueTM I like to use the standard names used by the library we are **wrapping**, but the idea is for `clj-rq` **to be simple and inclusive**.

Even if someone doesn't know the names of Redis functions but knows about queues, they should be able to use `clj-rq`. In other words, it's important to keep `push!` and `pop!` calling their respective functions from `jedis`. These names are explicit about what is expected in execution.

Reply:
@avelino Using macros puts a couple of limitations on ourselves. Of course what you're asking is feasible, but I don't think that parsing directions, and adding them into the function definition is within our time budget.
ourselves
It would involve changing the way we load the parameters from a given class, which could dismantle our macro. And for goods sake, writing that shivers my whole spine. But it's feasible nevertheless.

I'll scratch out a way to have the direction parsed when loading the parameters, as I'm currently implementing the static documentation, as layed out [here](https://github.com/moclojer/clj-rq/pull/18#discussion_r1717411424).

Reply:
@avelino not just about the directions, in some functions we have things like a position (alike **before** and **after)** and some extra opts to add. If we straight forward make it "static" on this way, it would be easier to test out, since the tests we need to write involves some data transformation to fit the Redis patterns. But how can we improve it to maintain as dynamic as possible, @J0sueTM ?

Reply:
find out we can call specific opts while on code like this:
![image](https://github.com/user-attachments/assets/06b39a49-e713-433e-85a0-2f2fdff147b1)

Reply:
@avelino @Felipe-gsilva There are too many variables to keep updating dynamically. I still think that we can of course check the direction from l<foobar> or r<foobar>, and check the method parameters and parse them appropiately. It just will take a lot more time.

Reply:
@J0sueTM I don't know if I understood the last comment - I imagine it's about @Felipe-gsilva's comment

It's clear my proposal to keep `push!` and `pop!` as they were before and have them use the macro implementation, right?

Reply:
@avelino Yes. My comment was regarding Felipe's coment. This conversation all happened before our conversation, when I finally understood what you meant with

>  keep push! and pop! as they were before and have them use the macro implementation

Reply:
It's worth writing by sharing what we will do in a separate comment so it doesn't get lost here

https://github.com/moclojer/clj-rq/pull/18#discussion_r1719644011